pkg/reader/ccip.go

```diff
@@ -550,45 +550,116 @@ func (r *ccipChainReader) GetChainFeePriceUpdate(ctx context.Context, selectors
 	return feeUpdates
 }
 
-func (r *ccipChainReader) DiscoverContracts(
+func (r *ccipChainReader) discoverDestinationContracts(
 	ctx context.Context,
-	destChain cciptypes.ChainSelector,
+	allChains []cciptypes.ChainSelector,
 ) (ContractAddresses, error) {
-	if err := validateExtendedReaderExistence(r.contractReaders, destChain); err != nil {
-		return nil, err
+	// Exit without an error if we cannot read the destination.
+	if err := validateExtendedReaderExistence(r.contractReaders, r.destChain); err != nil {
+		return nil, fmt.Errorf("validate extended reader existence: %w", err)
 	}
 
-	chains := maps.Keys(r.contractReaders)
+	// build up resp as we go.
+	var resp ContractAddresses
 
-	// OnRamps are in the offramp SourceChainConfig.
-	sourceConfigs, err := r.getSourceChainsConfig(ctx, chains)
+	// OnRamps are in the offRamp SourceChainConfig.
+	sourceConfigs, err := r.getOffRampSourceChainsConfig(ctx, allChains)
```

Comment:
took me some time to get why we need the `allChains` here. Maybe add a doc to the function.

Reply:
Updated comment on DiscoverContract. But also, this should hopefully go away soon.